.github/workflows/build.yml

```diff
@@ -0,0 +1,78 @@
+# yaml-language-server: $schema=https://json.schemastore.org/github-workflow.json
+name: Build Node
+
+on:
+  push:
+    branches:
+      - 'master'
+      - 'ibtida'
+
+    tags:
+      - 'v*'
+
+  pull_request:
+    branches:
+      - 'master'
+      - 'ibtida'
+
+jobs:
+  build-docker-images:
+    runs-on: ubuntu-20.04
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Get changes on Dockerfile
+        id: changed-dockerfile
+        uses: tj-actions/changed-files@v1.0.0
+        with:
+          files: Dockerfile*
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+        if: ${{ steps.changed-dockerfile.outputs.any_changed == 'true' || github.event_name != 'pull_request' }}
+
+      - name: Login to DockerHub
+        if: ${{ steps.changed-dockerfile.outputs.any_changed == 'true' || github.event_name != 'pull_request' }}
+        uses: docker/login-action@v1
+        with:
+          username: ${{ secrets.DOCKER_USER_NAME }}
+          password: ${{ secrets.DOCKER_PASSWORD }}
+
+      - name: Build and push builder image
+        uses: docker/build-push-action@v2
+        if: ${{ steps.changed-dockerfile.outputs.any_changed == 'true' }}
+        with:
+          file: ./Dockerfile.build
+          push: ${{ github.event_name != 'pull_request' }}
```

Comment:
Why do we include these `github.event_name != 'pull_request'` checks? Is it possible to get the same affect by not running this on pull request i.e. remove lines 13-16?

Reply:
Those checks will avoid pushing the image every time it's changed while the PR is open, but only when the PR is approved and merged into the main branches. However, if the Dockerfile is changed (action in lines 26-30), we still want to validate that it's building correctly.